pkg/daemon/ceph/osd/daemon.go

```diff
@@ -84,7 +88,66 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		if err := releaseLVMDevice(context, volumeGroupName); err != nil {
 			return fmt.Errorf("failed to release device from lvm. %+v", err)
 		}
+
+	}
+
+	return nil
+}
+
+func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) error {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigc:
+			logger.Infof("shutdown signal received, exiting...")
+			err := killCephOSDProcess(context, lvPath)
+			if err != nil {
+				return fmt.Errorf("failed to kill ceph-osd process. %+v", err)
+			}
+			if err := releaseLVMDevice(context, volumeGroupName); err != nil {
+				return fmt.Errorf("failed to release device from lvm. %+v", err)
+			}
+			return nil
+		}
+	}
+}
+
+func killCephOSDProcess(context *clusterd.Context, lvPath string) error {
+
+	processKilled := false
+	pid, err := context.Executor.ExecuteCommandWithOutput(false, "", "fuser", "-a", lvPath)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve process ID for - %s. Error %+v", lvPath, err)
 	}
+
+	logger.Debugf("process ID for ceph-osd: %s", pid)
+
+	// shut down the osd-ceph process so that lvm release does not show device in use error.
+	if pid != "" {
+		if err := context.Executor.ExecuteCommand(false, "", "kill", pid); err != nil {
+			return fmt.Errorf("failed to delete ceph-osd process. %+v", err)
+		}
+	} else {
+		return nil
+	}
+
+	pidInt, err := strconv.Atoi(pid)
+	if err != nil {
+		return fmt.Errorf("failed to convert process ID - %s to string. Error %+v ", pid, err)
+	}
+	for !processKilled {
```

Comment:
Even though, the ceph-osd should get the signal and exit, I'm always a bit hesitant with forever loops...

Reply:
well, as `FindProcess` will always succeed (according to the [docs](https://godoc.org/os#FindProcess)) this should not be an issue.
If I'm really motivated I'll whip up a patch that fixes this and uses Process.Signal instead of shelling out to `kill` as done above.

Reply:
Thanks for these inputs. I'll next be working on [3935](https://github.com/rook/rook/issues/3935) and see if workaround of killing ceph-osd process can be removed. I'll keeps these points in my mind while working on this PR.